packages/cozy-device-helper/withHasSafariPlugin.jsx

```diff
@@ -0,0 +1,31 @@
+import React, { Component } from 'react'
+import { isIOSApp, hasSafariPlugin } from '.'
+
+export const withHasSafariPlugin = eventHandlers => {
+  return WrappedComponent => {
+    return class withHasSafariPluginComponent extends Component {
+      state = {
+        hasSafariPlugin: false
+      }
+
+      checkSafariPlugin = async () => {
+        if (isIOSApp) {
+          const checked = await hasSafariPlugin()
+          if (checked !== this.state.hasSafariPlugin) {
```

Comment:
Is this to avoid rendering if nothing changed ? If yes, you can remove this and extend a [`PureComponent`](https://reactjs.org/docs/react-api.html#reactpurecomponent)